openstack_controller/tests/test_unit_glance.py

```diff
@@ -257,3 +259,80 @@ def test_images_metrics(aggregator, check, dd_run_check, metrics):
             tags=metric['tags'],
             hostname=metric.get('hostname'),
         )
+
+
+@pytest.mark.parametrize(
+    ('instance', 'paginated_limit', 'api_type', 'expected_api_calls', 'metrics'),
+    [
+        pytest.param(
+            configs.REST,
+            1,
+            ApiType.REST,
+            2,
+            IMAGES_METRICS_GLANCE,
+            id='api rest small limit',
+        ),
+        pytest.param(
+            configs.REST,
+            1000,
+            ApiType.REST,
+            1,
+            IMAGES_METRICS_GLANCE,
+            id='api rest high limit',
+        ),
+        pytest.param(
+            configs.SDK,
+            1,
+            ApiType.SDK,
+            1,
+            IMAGES_METRICS_GLANCE,
+            id='api sdk small limit',
+        ),
+        pytest.param(
+            configs.SDK,
+            1000,
+            ApiType.SDK,
+            1,
+            IMAGES_METRICS_GLANCE,
+            id='api sdk high limit',
+        ),
+    ],
+)
+@pytest.mark.usefixtures('mock_http_get', 'connection_image', 'mock_http_post', 'openstack_connection')
+def test_images_pagination(
+    aggregator,
+    instance,
+    openstack_controller_check,
+    paginated_limit,
+    expected_api_calls,
+    api_type,
+    dd_run_check,
+    mock_http_get,
+    connection_image,
+    metrics,
+):
+    paginated_instance = copy.deepcopy(instance)
+    paginated_instance['paginated_limit'] = paginated_limit
+    dd_run_check(openstack_controller_check(paginated_instance))
+    if api_type == ApiType.REST:
+        args_list = []
+        for call in mock_http_get.call_args_list:
+            args, kwargs = call
+            args_list += list(args)
+            params = kwargs.get('params', {})
+            limit = params.get('limit')
+            args_list += [(args[0], limit)]
+        assert args_list.count(('http://127.0.0.1:9292/image/v2/images', paginated_limit)) == expected_api_calls
+    else:
+        assert connection_image.images.call_count == 1
+        assert connection_image.images.call_args_list.count(mock.call(limit=paginated_limit)) == 1
+    for metric in metrics:
+        aggregator.assert_metric(
+            metric['name'],
+            metric_type=metric['metric_type'],
+            count=metric['count'],
+            value=metric['value'],
+            tags=metric['tags'],
+            hostname=metric.get('hostname'),
+            flush_first_value=metric.get('flush_first_value'),
```

Comment:
Why are we asserting the flush_first_value param?

Reply:
I initially included a misformatted image test file, which seems to have required changing the value of value, as well as asserting the "metric_type" and "flush_first_value" parameter in order to pass pre-existing test cases. However, in addressing Sarah's comment through commit [54b8ce0](https://github.com/DataDog/integrations-core/pull/16976/commits/54b8ce0a2e1f0621486639ddd1839855d1265e3a), these changes were now unnecessary, so I have undone them.